au/magnitude.hh

```diff
@@ -316,10 +316,101 @@ constexpr MagRepresentationOrError<T> checked_int_pow(T base, std::uintmax_t exp
     return result;
 }
 
-template <typename T, std::intmax_t N, typename B>
+template <typename T>
+constexpr MagRepresentationOrError<T> root(T x, std::uintmax_t n) {
+    // The "zeroth root" would be mathematically undefined.
+    if (n == 0) {
+        return {MagRepresentationOutcome::ERR_INVALID_ROOT};
+    }
+
+    // The "first root" is trivial.
+    if (n == 1) {
+        return {MagRepresentationOutcome::OK, x};
+    }
+
+    // We only support nontrivial roots of floating point types.
+    if (!std::is_floating_point<T>::value) {
```

Comment:
I initially had it as one, but then I realized that there are some values of `n` and `x` that make sense for integral types.  Granted, they're all trivial.  :grin:  But since we're returning a value-or-error type instead of a simple value, I like being able to communicate this --- especially since it fits so nicely into a pre-existing error category.

I would also worry more about this kind of thing if we intended to expose `root` externally.